File: dev-redo/packages/command/command/src/model/command/Light/LightOffCommand.ts
import ICommand from '../../../interface/ICommand';
import Light from '.';

class LightOffCommand implements ICommand {
  light: Light;
  constructor(light: Light) {
    this.light = light;
  }
  execute(): void {
    this.light.off();
  }
}

export default LightOffCommand;

File: jiggyjiggy/command/example/many/button/LightOffCommand.java
package command.example.many.button;

public class LightOffCommand implements Command {
	private Light light;
	
	public LightOffCommand(Light light) {
		this.light = light;
	}
	
	@Override
	public void execute() {
		light.off();
	}
	
	@Override
	public void undo() {
	
	}
}
